application/src/main/kotlin/net/corda/v5/application/crypto/SigningService.kt

```diff
@@ -36,4 +36,7 @@ interface SigningService {
      */
     @Suspendable
     fun sign(bytes: ByteArray, publicKey: PublicKey, signatureSpec: SignatureSpec): DigitalSignature.WithKey
+
+    @Suspendable
+    fun sign(bytes: ByteArray, publicKey: PublicKey): DigitalSignatureWithSpec
```

Comment:
There was some discussion on this today. There  are some tradeoffs with this new operation:

For safe operation, the trouble is that the signed payload must contain the signature spec, but the caller doesn't know the signature spec until we get the result. The implementation _could_ just choose the signature spec, then append it to the bytes, then sign that. But then it's no longer true that we are signing the `bytes` argument, and the recipient wil need to split out the payload from the signature spec. 

Consider a caller that wants to send messages as JSON. It wouldn't be valid JSON to just append a serialised signtature spec to the end of the payload.

This also means that some kind of serialization - of the signature spec, and how we combined it with the bytes payload, must be done by the crypto code with no control from the caller.

Further, this means that the signature for a given keypiar is not going to match what other implementations would produce, when signing `bytes` itself. It means that no other implementation could exist without assuming the internal serialisation details of the crypto worker, e.g. appended AMQP for the signing spec.